src/engraving/dom/edit.cpp

```diff
@@ -5897,47 +5897,71 @@ void Score::undoAddElement(EngravingItem* element, bool addToLinkedStaves, bool
                     const bool linkedPart  = linked && (staff != ostaff) && (staff->score() == ostaff->score());
                     const bool linkedScore = linked && (staff != ostaff) && (staff->score() != ostaff->score());
                     if (linkedPart && !linkedScore) {
-                        tr.push_back(staff->idx() * VOICES + mu::value(mapping, track) % VOICES);
+                        track_idx_t mappedTrack = mu::value(mapping, track, mu::nidx);
+                        if (mappedTrack == mu::nidx) {
+                            continue;
+                        }
+                        track_idx_t linkedTrack = staff->idx() * VOICES + mappedTrack % VOICES;
+                        if (!staff->isVoiceVisible(strack % VOICES)) {
+                            continue;
+                        }
+                        tr.push_back(linkedTrack);
                     } else if (!linkedPart && linkedScore) {
+                        // Staff in linked score
                         if ((track >> 2) != staffIdx) {
+                            // Element on linked staff in linked part
                             track += (staffIdx - (track >> 2)) * VOICES;
+                            if (!staff->isVoiceVisible(strack % VOICES)) {
+                                continue;
+                            }
                         }
                         tr.push_back(track);
                     } else {
+                        // Staff in part edit was performed on
                         tr.push_back(track);
                     }
                 }
             }
         }
 
         // Some elements in voice 1 of a staff should be copied to every track which has a linked voice in this staff
-
-        if (tr.empty() && (element->isSymbol()
-                           || element->isImage()
-                           || element->isTremoloBar()
-                           || element->isDynamic()
-                           || element->isExpression()
-                           || element->isStaffText()
-                           || element->isPlayTechAnnotation()
-                           || element->isCapo()
-                           || element->isStringTunings()
-                           || element->isSticking()
-                           || element->isFretDiagram()
-                           || element->isHarmony()
-                           || element->isHairpin()
-                           || element->isOttava()
-                           || element->isTrill()
-                           || element->isSlur()
-                           || element->isVibrato()
-                           || element->isTextLine()
-                           || element->isPedal()
-                           || element->isLyrics())) {
+        voice_idx_t voice = track2voice(strack);
+        if (staff->isVoiceVisible(voice) && tr.empty() && (element->isSymbol()
```

Comment:
It would be great to simplify this condition as it would be very hard to debug it